spec/ruby/spec_helper.rb

```diff
@@ -3,6 +3,9 @@
 dir = "fixtures/code"
 CODE_LOADING_DIR = use_realpath ? File.realpath(dir, root) : File.expand_path(dir, root)
 
+# Don't run ruby/spec as root
+raise 'ruby/spec is not designed to be run as root' if Process.uid == 0
```

Comment:
Any specific reason?

There are actually specs using `as_superuser` which only run as root, so it's somewhat designed to be run as `root`, but I would only do so in a container for safety.
I know some MRI committers checked that running the specs as root worked.

Reply:
I've found that most of it breaks as root - I'll try it again and report back.

The reason it's a useful check (if it does break as root) as that a few times we were accidentally running as root and it was confusing.

Reply:
@eregon what do you get if you run specs as root using `su`? I see lots of errors canonicalising paths and loading files.

Reply:
Did you try to run the specs with MRI or with TruffleRuby?

Reply:
I tried in Docker with Ubuntu 18.04 on MRI and almost all specs pass:
https://gist.github.com/eregon/1f826370e92612cef645072396d9eaca
The few failures seem due to an older Ruby version (2.5.1).

So on MRI, it's reasonable to run ruby/spec as root and actually needed for testing `as_superuser` specs. I should test running specs as root in https://github.com/ruby/spec to make sure they work.

I agree running specs as root on TruffleRuby is unintended, so how about moving this line to `spec/truffle.mspec` and adjusting the message?

Reply:
Done.